### ISSUE TITLE
Empty GradCAM++

### ISSUE THREAD

**Issue by javierrodenas** [totti0223/gradcamplusplus#3]
Hi!!

I have been testing with my own model using InceptionV3 + Food101 dataset

The Grad-CAM results are as expected . Otherwise, Grad-CAM++ returns the original image. Have you any idea about why it is returning nothing? 

I have seen that grad_CAM_map:
`grad_CAM_map = np.sum(deep_linearization_weights*conv_output[0], axis=2)`

returns:
```
[[nan nan nan nan nan]
 [nan nan nan nan nan]
 [nan nan nan nan nan]
 [nan nan nan nan nan]
 [nan nan nan nan nan]]
```

However, deep_linearization_weights and conv_output[0] are not empty. 

Thank you in advance

**Comment by totti0223** [comment]
Have you tried the native example codes in the repository in the first place?
If it doesn't run as the expected output, it may be caused by the Keras or TF version, as this was written a while ago.
Or else it may be an issue of yours.
If the later, pasting codes of your model building codes here might help.

**Comment by javierrodenas** [comment]
Hi @totti0223,

Ye, I have tried the native examples and it worked fine.

I am using keras 2.2.4 and tensorflow 1.15.

Could be that it is only prepared for VGG? As I told you, I am using Inception V3.

There are the last layers of my model:

__________________________________________________________________________________________________
batch_normalization_93 (BatchNo (None, 5, 5, 384)    1152        conv2d_93[0][0]                  
__________________________________________________________________________________________________
conv2d_94 (Conv2D)              (None, 5, 5, 192)    393216      average_pooling2d_9[0][0]        
__________________________________________________________________________________________________
batch_normalization_86 (BatchNo (None, 5, 5, 320)    960         conv2d_86[0][0]                  
__________________________________________________________________________________________________
activation_88 (Activation)      (None, 5, 5, 384)    0           batch_normalization_88[0][0]     
__________________________________________________________________________________________________
activation_89 (Activation)      (None, 5, 5, 384)    0           batch_normalization_89[0][0]     
__________________________________________________________________________________________________
activation_92 (Activation)      (None, 5, 5, 384)    0           batch_normalization_92[0][0]     
__________________________________________________________________________________________________
activation_93 (Activation)      (None, 5, 5, 384)    0           batch_normalization_93[0][0]     
__________________________________________________________________________________________________
batch_normalization_94 (BatchNo (None, 5, 5, 192)    576         conv2d_94[0][0]                  
__________________________________________________________________________________________________
activation_86 (Activation)      (None, 5, 5, 320)    0           batch_normalization_86[0][0]     
__________________________________________________________________________________________________
mixed9_1 (Concatenate)          (None, 5, 5, 768)    0           activation_88[0][0]              
                                                                 activation_89[0][0]              
__________________________________________________________________________________________________
concatenate_2 (Concatenate)     (None, 5, 5, 768)    0           activation_92[0][0]              
                                                                 activation_93[0][0]              
__________________________________________________________________________________________________
activation_94 (Activation)      (None, 5, 5, 192)    0           batch_normalization_94[0][0]     
__________________________________________________________________________________________________
mixed10 (Concatenate)           (None, 5, 5, 2048)   0           activation_86[0][0]              
                                                                 mixed9_1[0][0]                   
                                                                 concatenate_2[0][0]              
                                                                 activation_94[0][0]              
__________________________________________________________________________________________________
average_pooling2d_10 (AveragePo (None, 2, 2, 2048)   0           mixed10[0][0]                    
__________________________________________________________________________________________________
dropout_1 (Dropout)             (None, 2, 2, 2048)   0           average_pooling2d_10[0][0]       
__________________________________________________________________________________________________
flatten_1 (Flatten)             (None, 8192)         0           dropout_1[0][0]                  
__________________________________________________________________________________________________
dense_1 (Dense)                 (None, 27)           221211      flatten_1[0][0]                  

And I am running the methods as following:

```
gradcam=grad_cam(model,img,layer_name='conv2d_94')
gradcamplus=grad_cam_plus(model,img,layer_name='conv2d_94')
```
Is it ok to use **conv2d_94** layer? It is the last conv layer I saw...

Do you see any error? 

Thank you in advance!

**Comment by totti0223** [comment]
Hi,
As for your case, the last convolutional layer (literally not, but that contains the convoluted information corresponding for GRAD-CAMs) for InceptionV3 is the *mixed10 layer*, which is the end of the last inception module of feature extraction layers. I think conv2d_94 is a conv layer somewhere inside the inception module that does not reflect all the extracted features of the network. (search for a graphical summary of inceptionV3. should help)

**Comment by javierrodenas** [comment]
Hi @totti0223,

thank you very much for your response. I have been checking the structure of inceptionV3 and as you mentioned, the last conv layer is the mixed10layer. I don't know why it was printing conv2d_94 layer at the end of the network.

Thanks!
